indico/queries/custom_blueprint.py

```diff
@@ -0,0 +1,136 @@
+import typing as t
+
+from indico import GraphQLRequest
+from indico.errors import IndicoInputError
+from indico.types.workflow import ComponentFamily
+from indico.types.custom_blueprint import TaskBlueprint
+
+SUPPORTED_CUSTOM_COMPONENT_FAMILIES = [
+    ComponentFamily.OUTPUT,
+    ComponentFamily.FILTER,
+    ComponentFamily.MODEL,
+]
+
+
+class RegisterCustomBlueprint(GraphQLRequest):
+    """
+    Mutation to register a custom blueprint, making it available in the gallery to add to workflows
+
+    Args:
+        component_family (ComponentFamily): family this blueprint belongs to; supported families are Output, Filter, and Model
+        name (str): Name of the blueprint
+        description (str): Description of this blueprint
+        config (dict): Blueprint configuration options
+        tags: (list[str]): List of tags associated with this blueprint
+
+    Options:
+        footer (str): Footnote for this blueprint's description
+        icon (str): Name or storage location of image to use when displaying this blueprint in the gallery
```

Comment:
is the name the local path to the image?

Reply:
no, there are a small subset of icons that come automatically with each cluster, where you only have specify the name. There's a bunch more info here: https://developer.indicodata.ai/docs/registering-a-custom-blueprint#selecting-an-icon